package cms_8120;
import "type/cmd.proto";
option java_package = "com.hikvision.finance.fms.cms.hpp.center";
option java_outer_classname = "DeviceProto";
option java_generic_services = true;


//录像表
message Device
{
    //必须字段
    optional int32 id                       = 1;    // 在数据库中的流水号
    optional int32 nType                    = 2;    // 设备类型
    optional string strName                 = 3;    // 设备名称
    optional string strIp                   = 4;    // 设备IP地址
    optional int32 nPort                    = 5;    // 设备端口号
    optional string strUser                 = 6;    // 设备用户名
    optional string strPassword             = 7;    // 设备密码
    optional string strPasswordLevel        = 8;    // 设备密码等级
    optional int32 nDevType                 = 9;    // 产品类型，设备型号
    optional int32 nProductType             = 10;   // 设备型号
    optional int32 nChanNum                 = 11;   // 模拟通道个数
    optional string strDeviceId             = 12;   // 设备注册ID
    optional int32 telChanNum               = 13;   // ip通道个数
    optional int32 nRegionId                = 14;   // 所属区域流水号
    optional int32 nStreamType              = 15;   // 主子码流类型
    optional int32 dvrType                  = 16;   // 1:源DVR，2：采集DVR
    optional int32 nConfigPort              = 17;   // 设备配置端口
    optional string strConfigUser           = 18;   // 设备配置用户名
    optional string strConfigPassword       = 19;   // 设备配置密码
    optional int32 online                   = 20;   // 在线状态
    optional int32 uiTotalSpace             = 21;   // 硬盘总容量
    optional int32 uiFreeSpace              = 22;   // 硬盘空闲容量
    optional string strIndexCode            = 23;   // 设备编号
    optional string strManufacture          = 24;   // 厂商
    optional string strAccessKey            = 25;   //对象云存储AccessKey
    optional string strSecretKey            = 26;   //对象云存储SecretKey

}

message ReqDevice
{
    optional int32 cmd                	    = 1;    // CMD_DT_STORAGEDEVICE_INFO--CMD_DT_STORAGEDEVICE_LIST
    optional int32 nType                    = 2;    // 设备类型
    optional int32 id                       = 3;    // 在数据库中的流水号
    optional string strIp                   = 4;    // 设备IP地址
    optional int32 nPort                    = 5;    // 设备端口号
    optional int32 online                   = 6;    // 设备在线状态
    optional int32 uiTotalSpace             = 7;    // 硬盘总容量
    optional int32 uiFreeSpace              = 8;    // 硬盘空闲容量
}

message DeviceList
{
    optional int32 cmd                      = 1;    // CMD_DT_STORAGEDEVICE_INFO--CMD_DT_DVRDEVICE_LIST--CMD_DT_STORAGEDEVICE_LIST
    repeated Device device                  = 2;    // 设备列表
}


service ModifyDeviceService {
	
	rpc QueryStorageDeviceInfo(ReqDevice) returns (DeviceList);			// CMD_DT_STORAGEDEVICE_INFO(获取存储设备信息)
	
	rpc QueryStorageDeviceInfoBytype(ReqDevice) returns (DeviceList);	// CMD_DT_STORAGEDEVICE_LIST(根据设备类型获取存储设备信息)
	
	rpc QueryAllDvrInfo(ReqDevice) returns (DeviceList);				// CMD_DT_DVRDEVICE_LIST(获取所有DVR设备信息)
	
	
}

